pkg/apis/rke.cattle.io/v1/bootstrap.go

```diff
@@ -5,40 +5,81 @@ import (
 )
 
 // +genclient
-// +kubebuilder:skipversion
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=rkebootstraps,scope=Namespaced,categories=cluster-api
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels={"cluster.x-k8s.io/v1beta1=v1","auth.cattle.io/cluster-indexed=true"}
 
+// RKEBootstrap represents the BootstrapConfig resource that is required by CAPI to provide
+// the bootstrap data that is used to bootstrap a Kubernetes node.
+// It will be referenced by one of the Cluster API core resources, Machine.
+// More info: https://cluster-api.sigs.k8s.io/developer/providers/contracts/bootstrap-config
 type RKEBootstrap struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+
+	// Metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              RKEBootstrapSpec   `json:"spec"`
-	Status            RKEBootstrapStatus `json:"status,omitempty"`
+
+	// Spec is the desired state of RKEBootstrap.
+	Spec RKEBootstrapSpec `json:"spec"`
+
+	// Status is the observed state of RKEBootstrap.
+	// +optional
+	Status RKEBootstrapStatus `json:"status,omitempty"`
 }
 
+// RKEBootstrapSpec defines the desired state of RKEBootstrap.
 type RKEBootstrapSpec struct {
+	// ClusterName refers to the name of the CAPI Cluster associated with this RKEBootstrap.
+	// +optional
 	ClusterName string `json:"clusterName,omitempty"`
 }
 
+// RKEBootstrapStatus defines the observed state of RKEBootstrap.
 type RKEBootstrapStatus struct {
-	// Ready indicates the BootstrapData field is ready to be consumed
+	// Ready indicates the BootstrapData field is ready to be consumed.
+	// +optional
 	Ready bool `json:"ready,omitempty"`
 
 	// DataSecretName is the name of the secret that stores the bootstrap data script.
 	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
```

Comment:
Notes to reviewers: 
Validations are added to match the upstream: https://github.com/kubernetes-sigs/cluster-api/blob/main/api/bootstrap/kubeadm/v1beta1/kubeadmconfig_types.go#L459-L463